docs/docker_walkthrough.rst

```diff
@@ -0,0 +1,684 @@
+        - --core
+        - models
+        - -c
+        - rest
+        - --endpoints
+        - config/endpoints.yml
+        - -u
+        - current/
+    rasa_nlu:
+      image: rasa/rasa_nlu:latest-spacy
+      volumes:
+        - ./models/rasa_nlu:/app/models
+      command:
+        - start
+        - --path
+        - models
+
+
+3.3 Running Rasa Core and Rasa NLU
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To start Rasa Core and Rasa NLU execute:
+
+.. code-block:: bash
+
+  docker-compose up
+
+.. note::
+
+  Add the flag ``-d`` if you want to run it detached.
+
+The REST API of Rasa Core is then available on ``http://localhost:5005``.
+To send messages to your chatbot:
+
+.. code-block:: bash
+
+  curl --request POST \
+    --url http://localhost:5005/webhooks/rest/webhook \
+    --header 'content-type: application/json' \
+    --data '{
+      "message": "hello"
+    }'
+
+Your chatbot should then answer something like:
+
+.. code-block:: bash
+
+  [
+    {
+      "recipient_id": "default",
+      "text": "Hi, how is it going?"
+    }
+  ]
+
+If the chatbot cannot understand you, the answer is ``[]``.
+
+3.4 Adding a Custom NLU Pipeline
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to configure the components of your
+`NLU Pipeline <https://rasa.com/docs/nlu/choosing_pipeline/>`_, start by
+creating a file ``nlu_config.yml`` in your ``config`` directory.
+
+.. code-block:: bash
+
+  touch config/nlu_config.yml
+
+
+Put the description of your custom pipeline in there, e.g.:
+
+.. code-block:: yaml
+
+  pipeline:
+  - name: "nlp_spacy"
+  - name: "tokenizer_spacy"
+  - name: "intent_entity_featurizer_regex"
+  - name: "intent_featurizer_spacy"
+  - name: "ner_crf"
+  - name: "intent_classifier_sklearn"
+
+Then retrain your NLU model. In contrast to the previous training also mount
+the ``config`` directory which contains the NLU configuration
+and specify it in the run command.
+
+.. code-block:: bash
+
+  docker run \
+    -v $(pwd):/app/project \
+    -v $(pwd)/models/rasa_nlu:/app/models \
+    -v $(pwd)/config:/app/config \
+    rasa/rasa_nlu:latest-spacy \
+    run \
+      python -m rasa_nlu.train \
+      -c config/nlu_config.yml \
+      -d project/data/nlu.md \
+      -o models \
+      --project current
+
+Then adapt the NLU start command in your docker compose so that it uses your
+NLU configuration. As in for the training mount the ``config`` directory into
+your NLU container and instruct Rasa NLU to use this configuration by adding
+the flag ``-c <path to your nlu config>``.
+The configuration of the ``rasa_nlu`` server might then look similar to this:
+
+.. code-block:: yaml
+
+  rasa_nlu:
+      image: rasa/rasa_nlu:latest-spacy
+      volumes:
+        - ./models/rasa_nlu:/app/models
+        - ./config:/app/config
+      command:
+        - start
+        - --path
+        - models
+        - -c
+        - config/nlu_config.yml
+
+.. note::
+
+  Depending on the selected pipeline you might have to use a different
```

Comment:
Where a user can find more information about that? Would be good to add a link or something, because people do you their own custom configs very often.